spec/requests/admin/reports_controller_spec.rb

```diff
@@ -1,8 +1,8 @@
 # frozen_string_literal: true
 
 RSpec.describe Admin::ReportsController do
-  it "is a subclass of AdminController" do
-    expect(Admin::ReportsController < Admin::AdminController).to eq(true)
+  it "is a subclass of StaffController" do
```

Comment:
I know this is an existing spec we wrote but I don't really agree that it fits into a request spec. My vote will be to just drop it and have a test that ensures normal users cannot access the controller endpoint.

Reply:
I'd also vote for dropping specs that check inheritance hierarchy (which is an implementation detail) in favor of testing behavior, just as @tgxworld suggested.

This is definitely something that we can do at some point later, not necessary in this PR.

Reply:
Since the inheritance specs are tangentially related to the refactor, I'll include the changes in this PR.